consensus/hotstuff/voter/voter.go

```diff
@@ -45,6 +55,16 @@ func (v *Voter) ProduceVoteIfVotable(block *model.Block, curView uint64) (*model
 		return nil, model.NoVoteError{Msg: "not above the last voted view"}
 	}
 
+	// Do not produce a vote for blocks where we are not a valid committee
+	// member. These votes can't be used to produce valid QCs.
```

Comment:
```suggestion
	// member. HotStuff will ask for a vote when cross the epoch boundary. 
	// These votes can't be used to produce valid QCs.
```

Reply:
I don't quite understand this. HotStuff asks `Voter` for a vote for all blocks, both before and after the epoch boundary.

Reply:
good point. updated.